Modules/Office365DSC/DSCResources/MSFT_EXOAcceptedDomain/MSFT_EXOAcceptedDomain.psm1

```diff
@@ -0,0 +1,269 @@
+function Get-TargetResource
+{
+    [CmdletBinding()]
+    [OutputType([System.Collections.Hashtable])]
+    param
+    (
+        [Parameter()]
+        [ValidateSet('Authoritative')]
+        [System.String]
+        $DomainType = 'Authoritative',
+
+        [Parameter()]
+        [ValidateSet('Present', 'Absent')]
+        [System.String]
+        $Ensure = 'Present',
+
+        [Parameter(Mandatory = $true)]
+        [System.Management.Automation.PSCredential]
+        $GlobalAdminAccount,
+
+        [Parameter(Mandatory = $true)]
+        [ValidatePattern( '(?=^.{1,254}$)(^(?:(?!\d+\.|-)[a-zA-Z0-9_\-]{1,63}(?<!-)\.?)+(?:[a-zA-Z]{2,})$)' )]
+        [System.String]
+        $Identity,
+
+        [Parameter()]
+        [ValidateScript( {$false -eq $_})]
+        [System.Boolean]
+        $MatchSubDomains = $false,
+
+        [Parameter()]
+        [ValidateScript( {$false -eq $_})]
+        [System.Boolean]
+        $OutboundOnly = $false
+    )
+    Write-Verbose "Get-TargetResource will attempt to retrieve Accepted Domain configuration for $($Identity)"
+    Write-Verbose "Calling Connect-ExchangeOnline function:"
+    Connect-ExchangeOnline -GlobalAdminAccount $GlobalAdminAccount
+    Write-Verbose "Global ExchangeOnlineSession status:"
+    Write-Verbose "$( Get-PSSession -ErrorAction SilentlyContinue | Where-Object Name -eq 'ExchangeOnline' | Out-String)"
+    try
+    {
+        $AllAcceptedDomains = Get-AcceptedDomain
+    }
+    catch
+    {
+        Close-SessionsAndReturnError -ExceptionMessage $_.Exception
+    }
+
+    $AcceptedDomain = ($AllAcceptedDomains | Where-Object Identity -IMatch $Identity)
+
+    if (!$AcceptedDomain)
+    {
+        Write-Verbose "AcceptedDomain configuration for $($Identity) does not exist."
+
+        # Check to see if $Identity matches a verified domain in the O365 Tenant
+        try
+        {
+            Test-O365ServiceConnection -GlobalAdminAccount $GlobalAdminAccount
+            $VerifiedDomains = (Get-AzureADDomain | Where-Object IsVerified)
+            $MatchingVerifiedDomain = $VerifiedDomains | Where-Object Name -eq $Identity
+        }
+        catch
+        {
+            Close-SessionsAndReturnError -ExceptionMessage $_.Exception
+        }
+
+        if (-NOT $MatchingVerifiedDomain)
+        {
+            Write-Verbose "A verified domain matching $($Identity) does not exist in this O365 Tenant."
+            $nullReturn = @{
+                DomainType         = $DomainType
+                Ensure             = $Ensure
```

Comment:
Note comments in that block.  If Absent is returned there it will result in a call to Set-TargetResource, which will inevitably throw an error since there is no verified domain match.  That's fine for end-users, but will that cause issues for ReverseDsc?

Reply:
Gotcha, no it should cause issues for the Export-TargetResource. Thanks